### PR TITLE
Serialize contract with borsh

### PR DESCRIPTION
**Pull request by volovyks**

### CODE REVIEW

**Comment by volovyks** [comment]
@ChaoticTempest it is failing with `status: Failure(ActionError(ActionError { index: Some(0), kind: FunctionCallError(ExecutionError("Smart contract panicked: panicked at contract/src/lib.rs:215:1:\nFailed to deserialize input from JSON.: Error(\"expected value\", line: 1, column: 1)")) })),`.
Seams like I can not convince it that we are using borsh.

**Comment by volovyks** [comment]
We can choose [another approach](https://github.com/near-examples/factory-rust/blob/b70459cdb53bde5f041fd33110a872c2bcc3b5a4/src/manager.rs#L12), read it as env::input(), not a function parameter.

[CLI example](https://github.com/near-examples/factory-rust?tab=readme-ov-file#update-the-stored-contract). 

[Example from sputnik](https://github.com/near-daos/sputnik-dao-contract/blob/640495ba572345ca356376989738fbd5462e1ff8/sputnikdao2/src/upgrade.rs#L130-L140).
Thanks @gagdiez 

I have not looked yet, but Illya suggested taking a look at the [Aurora contract](https://github.com/Near-One/rainbow-token-connector/blob/master/token-locker/src/lib.rs#L70).

**Comment by ChaoticTempest** [comment]
@volovyks so I tested how much gas a 4mb payload (the limit of a transaction) would take. It's about 93TGas, so we at least have 200TGas to use for the rest of the function which is just storing the contract code anyways.

It doesn't have to be this efficient quite yet plus this is just an interface the node will be using so we can change it at will

**Comment by ChaoticTempest** [comment]
also, not sure why the multichain-contract test is not running. It should be listening into the the files changes under `chain-signature/contract/**`

**Comment by ChaoticTempest** [comment]
huh the multichain contract workflow was disabled for some reason

**Review by ChaoticTempest — APPROVED** [review]

File: chain-signatures/contract/tests/tests.rs [hunk 1]
```diff
@@ -499,18 +523,15 @@ async fn test_propose_update_config(contract: &Contract, accounts: &[Account]) {
...
         contract.view("state").await.unwrap().json().unwrap();
 
     // Let's propose a contract update instead now.
-    let new_wasm = std::fs::read(CONTRACT_FILE_PATH).unwrap();
     let execution = accounts[0]
         .call(contract.id(), "propose_update")
-        .args_json(serde_json::json!({
-            "code": &new_wasm,
-        }))
+        .args_borsh((current_contract(),))
```
Comment (volovyks) [inline chain-signatures/contract/tests/tests.rs:534]: @ChaoticTempest how do additional brackets help here? :)

Comment (ChaoticTempest) [reply chain-signatures/contract/tests/tests.rs:534]: Borsh serialization expects an iterable list of args. Adding `(T,)` makes it into a tuple type that is iterable

**Pull request merged**